thirdeye-server/src/main/java/ai/startree/thirdeye/alert/AlertEvaluator.java

```diff
@@ -88,6 +89,8 @@ public AlertEvaluationApi evaluate(final AlertEvaluationApi request)
           detectionInterval);
 
       final DetectionPipelineContext context = new DetectionPipelineContext()
+          .setAlertId(request.getAlert().getId())
+          .setUsage(DetectionPipelineUsage.EVALUATION)
```

Comment:
Just FYI. I think we are creating different names for the same thing. 

In `DetectionPipelineContext`
```  
private boolean preserveOutputDataFrames = false;
````

this variable is used to understand whether we need to preserve the big data frames.

Reply:
```
private boolean preserveOutputDataFrames = false;
```

yes saw this one. 
I could have used this boolean but the name was specific to the forkjoin logic, and I'm more confident with an enum, in the future we could allow things like:
EVALUATE_AND_DROP_DATAFRAMES (for some perf opti?)
or 
DETECT_KEEP_DATAFRAME (to store historical data in persistence/generate preview in notifications?)
DRIFT_DETECTION?
etc...

I'll drop `preserveOutputDataFrames` in favor of `usage` in another PR if it's okay for you.